table/block_based/block_based_table_builder.cc

```diff
@@ -1247,13 +1255,14 @@ void BlockBasedTableBuilder::WriteRawBlock(const Slice& block_contents,
     io_s = r->file->Append(Slice(trailer, kBlockTrailerSize));
     if (io_s.ok()) {
       assert(s.ok());
-      if (is_data_block &&
-          r->table_options.prepopulate_block_cache ==
-              BlockBasedTableOptions::PrepopulateBlockCache::kFlushOnly) {
+      if (r->table_options.prepopulate_block_cache ==
+          BlockBasedTableOptions::PrepopulateBlockCache::kFlushOnly) {
         if (type == kNoCompression) {
-          s = InsertBlockInCache(block_contents, handle);
+          s = InsertBlockInCache(block_contents, handle, block_entry,
```

Comment:
Do we insert all block types or just index/filter/data?

Reply:
@anand1976 It currently inserts all block types. I wasn't sure about other blocks  and their impact in terms of caching them so I inserted them also. Do we need them or they can be ignored? ```BlockBasedTableReader``` doesn't cache them and those blocks are used only once during opening the table so they can be ignored.

Reply:
@akankshamahajan15 They can be ignored, except for uncompression dict. I don't think ```BlockBasedTableReader``` even looks in the cache for other block types.